spark/datadog_checks/spark/spark.py

```diff
@@ -331,7 +331,22 @@ def _get_standalone_app_url(self, app_id, tags):
         app_detail_ui_links = dom.find_all('a', string='Application Detail UI')
 
         if app_detail_ui_links and len(app_detail_ui_links) == 1:
-            return app_detail_ui_links[0].attrs['href']
+            url = urlparse(app_detail_ui_links[0].attrs['href'])
+            # The href links can sometimes be a relative link. The helper functions combines it with the master_address
+            # in an attempt to collect metrics.
+            if not url.scheme and url.path:
+                return self._relative_link_transformer(self.master_address, url.path)
+            else:
+                return app_detail_ui_links[0].attrs['href']
+
+    def _relative_link_transformer(self, base_url, relative_url):
+        self.log.debug(
+            "Failed to find valid URL, assuming link is relative. "
+            "Combining relative path with master url: %s with path: %s",
+            base_url,
+            relative_url,
+        )
+        return urljoin(base_url, relative_url)
```

Comment:
I think I'd rather inline this in the function above and remove the test (as it is the test is purely testing `urljoin`, which we didn't implement).

If we want to have testable unit of code, I'd extract lines 334-340 (including an inlined version of `_relative_link_transformer`) into a separate method and test that both on cases where the url includes a scheme and where it doesn't (relative).